tests/phpunit/api/v3/OpenIDTest.php

```diff
@@ -0,0 +1,90 @@
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+require_once 'CiviTest/CiviUnitTestCase.php';
+
+
+/**
+ *  Test APIv3 civicrm_openid_* functions
+ *
+ * @package CiviCRM_APIv3
+ * @subpackage API_Contact
+ */
+class api_v3_OpenIDTest extends CiviUnitTestCase {
+
+  protected $_apiversion = 3;
+  protected $params;
+  protected $id;
+  protected $_entity;
+
+  public $DBResetRequired = FALSE;
+
+  public function setUp() {
+    parent::setUp();
+    $this->useTransaction(TRUE);
+
+    $this->_entity = 'openID';
+    $this->_contactID = $this->organizationCreate();
+    $this->params = array(
+      'contact_id' => $this->_contactID,
+      'name' => 'My OpenID handle',
+      'location_type_id' => 1,
+    );
+  }
+
+  public function testCreateOpenID() {
+    $result = $this->callAPIAndDocument($this->_entity, 'create', $this->params, __FUNCTION__, __FILE__);
+    $this->assertEquals(1, $result['count'], 'In line ' . __LINE__);
+    $this->getAndCheck($this->params, $result['id'], $this->_entity);
+    $this->assertNotNull($result['values'][$result['id']]['id'], 'In line ' . __LINE__);
+  }
+
+  public function testGetOpenID() {
+    $result = $this->callAPISuccess($this->_entity, 'create', $this->params);
+    $result = $this->callAPIAndDocument($this->_entity, 'get', $this->params, __FUNCTION__, __FILE__);
+    $this->assertEquals(1, $result['count'], 'In line ' . __LINE__);
+    $this->assertNotNull($result['values'][$result['id']]['id'], 'In line ' . __LINE__);
+    $this->callAPISuccess($this->_entity, 'delete', array('id' => $result['id']));
+  }
+
+  public function testDeleteOpenID() {
+    $result = $this->callAPISuccess($this->_entity, 'create', $this->params);
+    $deleteParams = array('id' => $result['id']);
+    $result = $this->callAPIAndDocument($this->_entity, 'delete', $deleteParams, __FUNCTION__, __FILE__);
+    $checkDeleted = $this->callAPISuccess($this->_entity, 'get', array());
+    $this->assertEquals(0, $checkDeleted['count'], 'In line ' . __LINE__);
+  }
+
+  public function testDeleteOpenIDInvalid() {
```

Comment:
Another aside. I think these 'failure tests' are at best marginal. The wrapper itself is pretty heavily tested so unless you have a specific bit of code to test I don't see value in the invalid data tests